### PR TITLE
Add Testing suite

### PR DESCRIPTION
**Pull request by ramesht007**
## Description
Add testing suite 

## Relevant Issue
#94 
#42 

## Affected Dependencies
None

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/RecoEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/RecoEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/RecoEdge/labels)
- [x] My changes are covered by tests

### CODE REVIEW

File: fedrec/python_executors/trainer.py [hunk 1]
```diff
@@ -101,8 +101,8 @@ def serialize(self):
             state_dict=state,
             model_preproc=self.model_preproc,
             storage=self.persistent_storage,
-            local_sample_number=self.local_sample_number,
-            local_training_steps=self.local_training_steps
+            local_sample_number = self.local_sample_number,
```
Comment (vkkhare) [inline fedrec/python_executors/trainer.py:104]: remove spaces

File: test/integration_test.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                 config: Dict) -> None:
+        super().__init__(config)
+
+    @property
+    def worker(self):
+        return Trainer(worker_index=0,
+                       config=self.config,
+                       logger=self.logger)
+
+    def TestTraining(self):
```
Comment (vkkhare) [inline test/integration_test.py:68]: `test_training_method`

File: test/integration_test.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            job_args=None,
+            job_kwargs=None
+        )
+        # check response message
+        if response.status:
+            worker_state = response.results
+            self.worker.load_worker(worker_state)
+            print(f"Worker state {response.get_worker_state()}")
+
+    def TestTesting(self):
```
Comment (vkkhare) [inline test/integration_test.py:83]: test_testing_method

File: test/integration_test.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            job_args=None,
+            job_kwargs=None
+        )
+        if response.status:
+            worker_state = response.results
+            self.worker.load_worker(worker_state)
+            print(f"Worker State {response.get_worker_state()}")
+
+
+class TestFedAvg(AbstractTester):
```
Comment (vkkhare) [inline test/integration_test.py:97]: TestAggregator

File: test/integration_test.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        self.out_neighbours = out_neighbours
+
+    @property
+    def worker(self):
+        return Aggregator(worker_index=0, config=config,
+                          logger=self.logger,
+                          in_neighbours=self.in_neighbours,
+                          out_neighbours=self.out_neighbours)
+
+    def testagg(self):
```
Comment (vkkhare) [inline test/integration_test.py:116]: test_aggregation

File: test/integration_test.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            job_args=None,
+            job_kwargs=None
+        )
+        # check response message
+        if response.status:
+            worker_state = response.results
+            self.worker.load_worker(worker_state)
+            print(f"Worker State {response.get_worker_state()}")
+
+    def testsampclient(self,
```
Comment (vkkhare) [inline test/integration_test.py:130]: test_sample_client

File: test/integration_test.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+
+if __name__ == "__main__":
+
+    with open("../configs/dlrm_fl.yml", 'r') as cfg:
+        config = yaml.load(cfg, Loader=yaml.FullLoader)
+
+    print(config['model'])
+    test_trainer = TestTrainer(config=config)
+    test_trainer.TestTesting()
```
Comment (vkkhare) [inline test/integration_test.py:152]: update here as well

File: test/test_serializer.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    """test JSONSerializer method
+    """
+    message_dict_response = JobResponseMessage(job_type, senderid, receiverid)
+    serilized_response_msg = JSONSerializer.serialize(message_dict_response)
+    response_msg = json.loads(serilized_response_msg)
+    assert response_msg['__type__'] == message_dict_response.__type__
+    assert response_msg['__data__'] == message_dict_response.__dict__
+
+
+def test_JSON_jobSubmitMessage_serialize(job_type, job_args,
```
Comment (vkkhare) [inline test/test_serializer.py:40]: small case all

File: test/test_serializer.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    """test generate_message_dict method
+    """
+    obj = JobSubmitMessage(job_type, job_args, job_kwargs,
+                           senderid, receiverid, workerstate)
+    dict = AbstractSerializer.generate_message_dict(obj)
+    assert dict['__type__'] == obj.__type__
+    assert dict['__data__'] == obj.__dict__
+
+
+def test_JSON_JobResponseMessage_serialize(job_type, senderid, receiverid):
```
Comment (vkkhare) [inline test/test_serializer.py:30]: small case all

File: test/test_serializer.py [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    message_dict_submit = JobSubmitMessage(job_type, job_args,
+                                           job_kwargs, senderid,
+                                           receiverid, workerstate)
+    serilized_submit_msg = JSONSerializer.serialize(message_dict_submit)
+    response_submit_msg = json.loads(serilized_submit_msg)
+    assert response_submit_msg['__type__'] == message_dict_submit.__type__
+    assert response_submit_msg['__data__'] == message_dict_submit.__dict__
+
+
+def test_JSON_JobSubmitMessage_deserialize(job_type, job_args, job_kwargs,
```
Comment (vkkhare) [inline test/test_serializer.py:52]: small case all

File: test/test_serializer.py [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    """
+    message = JobSubmitMessage(job_type, job_args,
+                               job_kwargs, senderid,
+                               receiverid, workerstate)
+    serilized_msg = JSONSerializer.serialize(message)
+    response__msg = JSONSerializer.deserialize(serilized_msg)
+    assert response__msg == message
+
+
+def test_JSON_JobResponseMessage_deserialize(job_type,
```
Comment (vkkhare) [inline test/test_serializer.py:64]: small caps all

**Review by vkkhare — CHANGES_REQUESTED** [review]

**Review by vkkhare — APPROVED** [review]
